Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/hibernate/AuditRepositoryDAO.java

```diff
@@ -161,4 +166,150 @@ public List queryAuditRepositoryOnCriteria(String eUserId, String ePatientId, Da
         }
         return queryList;
     }
+
+    /**
+     * This method does a query to database to get the Audit Log Messages based different options
+     *
+     * @param messageId
+     * @param relatesTo
+     * @param outcome
+     * @param startDate
+     * @param userId
+     * @param eventTypeList
+     * @param remoteHcidList
+     * @param endDate
+     * @return List
+     */
+    public List queryAuditViewer(String messageId, String relatesTo, Integer outcome, List<String> eventTypeList,
+        String userId, List<String> remoteHcidList, String startDate, String endDate) {
+
+        Session session = null;
+        List<AuditRepositoryRecord> queryList = null;
+        try {
+            SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+            session = sessionFactory.openSession();
+
+            LOG.info("Getting Record for Audit Viewer ");
+
+            // Build the criteria
+            Criteria queryCriteria = session.createCriteria(AuditRepositoryRecord.class);
+
+            if (NullChecker.isNotNullish(messageId) || NullChecker.isNotNullish(relatesTo)) {
```

Comment:
These if statements need to be refactored.  Should remove the top level check (relatesto or messageId don't equal null) and handle them separately.
Also, you have a check for if message id startwith or doesn't start with urn:uuid, that or will always be true.